terminal2048/lib/terminal2048/state.ex

```diff
@@ -0,0 +1,7 @@
+defmodule Terminal2048.State do
+  defstruct(
+    game_state: :initializing,
+    board: List.duplicate(nil, 4) |> List.duplicate(4),
```

Comment:
It's hard to tell without any implementation code, but I'm worried that a list-of-lists might be an awkward data structure to use to represent the board.